pkg/authorization/authorizer/scope/converter.go

```diff
@@ -318,45 +343,55 @@ func (e clusterRoleEvaluator) Describe(scope string) (string, string, error) {
 	return description, warning, nil
 }
 
-func (e clusterRoleEvaluator) ResolveRules(scope, namespace string, clusterPolicyGetter authorizationlister.ClusterPolicyLister) ([]authorizationapi.PolicyRule, error) {
+func (e clusterRoleEvaluator) ResolveRules(scope, namespace string, clusterRoleGetter rbaclisters.ClusterRoleLister) ([]rbac.PolicyRule, error) {
 	_, scopeNamespace, _, err := e.parseScope(scope)
 	if err != nil {
 		return nil, err
 	}
 
 	// if the scope limit on the clusterrole doesn't match, then don't add any rules, but its not an error
 	if !(scopeNamespace == authorizationapi.ScopesAllNamespaces || scopeNamespace == namespace) {
-		return []authorizationapi.PolicyRule{}, nil
+		return []rbac.PolicyRule{}, nil
 	}
 
-	return e.resolveRules(scope, clusterPolicyGetter)
+	return e.resolveRules(scope, clusterRoleGetter)
+}
+
+func has(set []string, value string) bool {
+	for _, element := range set {
+		if value == element {
+			return true
+		}
+	}
+	return false
 }
 
 // resolveRules doesn't enforce namespace checks
-func (e clusterRoleEvaluator) resolveRules(scope string, clusterPolicyGetter authorizationlister.ClusterPolicyLister) ([]authorizationapi.PolicyRule, error) {
+func (e clusterRoleEvaluator) resolveRules(scope string, clusterRoleGetter rbaclisters.ClusterRoleLister) ([]rbac.PolicyRule, error) {
 	roleName, _, escalating, err := e.parseScope(scope)
 	if err != nil {
 		return nil, err
 	}
 
-	policy, err := clusterPolicyGetter.Get("default")
+	role, err := clusterRoleGetter.Get(roleName)
 	if err != nil {
+		if kapierrors.IsNotFound(err) {
+			return nil, kapierrors.NewNotFound(authorizationapi.LegacyResource("clusterrole"), roleName)
```

Comment:
we can stop synthesizing the error now... we have a real error that matches the type and name we were unable to find :)